nemoguardrails/actions/llm/utils.py

```
@@ -74,6 +75,13 @@ async def llm_call(
     _log_prompt(prompt)
     chat_prompt = _ensure_chat_messages(prompt)
 
+    # Validate context length before sending to LLM
+    try:
+        validate_context_length(prompt, model_name=model_name or model.model_name)
+    except ContextLengthExceededError as e:
+        logger.error(f"Context length validation failed: {e}")
+        raise LLMCallException(e)
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)